cmd/healthcheck/framework/healthcheck.go

```diff
@@ -0,0 +1,445 @@
+	broker := &v1beta1.ClusterServiceBroker{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: h.brokername,
+		},
+		Spec: v1beta1.ClusterServiceBrokerSpec{
+			CommonServiceBrokerSpec: v1beta1.CommonServiceBrokerSpec{
+				URL: url,
+			},
+		},
+	}
+
+	glog.V(4).Infof("checking for Broker %v to be ready", broker.Name)
+	err = util.WaitForBrokerCondition(h.serviceCatalogClientSet.ServicecatalogV1beta1(),
+		broker.Name,
+		v1beta1.ServiceBrokerCondition{
+			Type:   v1beta1.ServiceBrokerConditionReady,
+			Status: v1beta1.ConditionTrue,
+		},
+	)
+	if err != nil {
+		return h.setError("broker not ready: %v", err.Error())
+	}
+
+	err = util.WaitForClusterServiceClassToExist(h.serviceCatalogClientSet.ServicecatalogV1beta1(), h.serviceclassID)
+	if err != nil {
+		return h.setError("service class not found: %v", err.Error())
+	}
+	return nil
+}
+
+// createInstance creates a Service Instance and verifies it becomes ready
+// and it's references are resolved
+func (h *HealthCheck) createInstance() error {
+	if h.frameworkError != nil {
+		return h.frameworkError
+	}
+	glog.V(4).Info("Creating a ServiceInstance")
+	instance := &v1beta1.ServiceInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      h.instanceName,
+			Namespace: h.namespace.Name,
+		},
+		Spec: v1beta1.ServiceInstanceSpec{
+			PlanReference: v1beta1.PlanReference{
+				ClusterServiceClassExternalName: h.serviceclassName,
+				ClusterServicePlanExternalName:  "default",
+			},
+		},
+	}
+	operationStartTime := time.Now()
+	var err error
+	instance, err = h.serviceCatalogClientSet.ServicecatalogV1beta1().ServiceInstances(h.namespace.Name).Create(instance)
+	if err != nil {
+		return h.setError("error creating instance: %v", err.Error())
+	}
+
+	if instance == nil {
+		return h.setError("error creating instance - instance is null", "")
+	}
+
+	glog.V(4).Info("Waiting for ServiceInstance to be ready")
+	err = util.WaitForInstanceCondition(h.serviceCatalogClientSet.ServicecatalogV1beta1(),
+		h.namespace.Name,
+		h.instanceName,
+		v1beta1.ServiceInstanceCondition{
+			Type:   v1beta1.ServiceInstanceConditionReady,
+			Status: v1beta1.ConditionTrue,
+		},
+	)
+	if err != nil {
+		return h.setError("instance not ready: %v", err.Error())
+	}
+	ReportOperationCompleted("create_instance", operationStartTime)
+
+	glog.V(4).Info("Verifing references are resolved")
+	sc, err := h.serviceCatalogClientSet.ServicecatalogV1beta1().ServiceInstances(h.namespace.Name).Get(h.instanceName, metav1.GetOptions{})
+	if err != nil {
+		return h.setError("error getting instance: %v", err.Error())
+	}
+
+	if sc.Spec.ClusterServiceClassRef == nil {
+		return h.setError("ClusterServiceClassRef should not be null", "")
+	}
+	if sc.Spec.ClusterServicePlanRef == nil {
+		return h.setError("ClusterServicePlanRef should not be null", "")
+	}
+
+	if strings.Compare(sc.Spec.ClusterServiceClassRef.Name, h.serviceclassID) != 0 {
+		return h.setError("ClusterServiceClassRef.Name should not be null", "")
```

Comment:
The error message here, and the next check, are copy/paste errors. They should output the expected value instead.